unliftio/test/UnliftIO/ExceptionSpec.hs

```diff
@@ -79,6 +79,27 @@ spec = do
       result <- withWrappedAsyncExceptionThrown $ \m -> trySyncOrAsync (void m)
       first fromExceptionUnwrap result `shouldBe` Left (Just Exception1)
 
+  describe "withException" $ do
+    it "should work when withAsync is in the handler" $ do
+      let
+        action =
+          error "oops"
+            `onException` do
+              let
+                timerAction n = do
+                  threadDelay 1000000
+                  when (n < 10) $ do
+                    timerAction (n + 1)
+              withAsync (timerAction 0) $ \a -> do
+                cancel a
```

Comment:
This line can be replaced with `pure ()`. `withAsync` is *also* unable to `cancel` the thread.